cmd/buildkitd/config/config.go

```diff
@@ -73,9 +72,9 @@ type OCIConfig struct {
 	ProxySnapshotterPath string `toml:"proxySnapshotterPath"`
 
 	// StargzSnapshotterConfig is configuration for stargz snapshotter.
-	// Decoding this is delayed in order to remove the dependency from this
-	// config pkg to stargz snapshotter's config pkg.
-	StargzSnapshotterConfig toml.Primitive `toml:"stargzSnapshotter"`
+	// We use a generic map[string]interface{} in order to remove the dependency
+	// on stargz snapshotter's config pkg from our config.
+	StargzSnapshotterConfig map[string]interface{} `toml:"stargzSnapshotter"`
```

Comment:
This was changed to a generic `map[string]interface{}`, which go-toml is able to Unmarshal into the desired type through `toml.TreeFromMap()`. This brings the additional benefit that we don't even need to import any toml package here.

I should note that I found that `github.com/pelletier/go-toml/v2` does _not_ have `toml.TreeFromMap()`, but I think it's able to handle it in different ways (I didn't spend much time on that yet; other dependencies are still on v1 as well, because v2 is still in beta).